### ISSUE TITLE
[Feature Request]: Align Figma component attribute names to React prop names

### ISSUE THREAD

**Issue by shixiedesign** [carbon-design-system/carbon#14836]
### The problem

Currently the props of the components are named differently in Figma and in Code. This defeats the goal of "design system being **the common language** between design and development". It's understandable that not all properties translate, but at least the visual properties should match. 

For example, button have `kind`, and I prefer it over `style`. 

<img width="1056" alt="image" src="https://github.com/carbon-design-system/carbon/assets/15144993/252b45c8-aecb-432e-a945-0cc9a2c17578">

### The solution

To have perfect alignment between these naming system is impossible, and could cause unnecessary changes to Figma component construction too. 

For now, how about just change button's "Style" to "Kind", and fix a few easy ones? 

### Business priority

None

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)

**Comment by tay1orjones** [comment]
@shixiedesign Yeah I agree, @aubrey-oneal did a big audit investigation into this. I think it's under an existing issue, https://github.com/carbon-design-system/carbon/issues/12947

It was previously blocked by us needing to upgrade to storybook 7 but we'd like to get these sorted out when we can.

Take a look there - if you think it encapsulates well the same thing you're proposing here, we can maybe close this one as a duplicate? Let me know

**Comment by tay1orjones** [comment]
We reviewed this issue as a team today and agree that this is part of https://github.com/carbon-design-system/carbon/issues/12947. I'll close this for now, but if there's anything else we're missing, please let us know.

**Comment by tay1orjones** [comment]
Duplicate of https://github.com/carbon-design-system/carbon/issues/12947
